### DISCUSSION TITLE
How to use FEM?

### DISCUSSION THREAD

**Discussion by VallesMarinerisExplorer** [bulletphysics/bullet3#3918] [General]
Hi all,

I saw that FEM was recently added to bullet physics in modeling soft bodies. Are there any examples for PyBullet or is there any guidance on how to integrate this kind of new functionality into the API to use with PyBullet?

Thanks!

**Comment by PeterQiu0516** [comment]
To enable FEM for deformables, you gotta reset the world as follows before simulation:
```python
pybullet.resetSimulation(p.RESET_USE_DEFORMABLE_WORLD)
```

You could have a look at this [example](https://github.com/bulletphysics/bullet3/blob/master/examples/pybullet/examples/deformable_anchor.py#L8), which involves a deformable cloth with 2 corners attached to free space and other 2 attached to a falling cube.

**Reply by yonkshi** [reply]
To add to what @PeterQiu0516 said, the example is a good starting point. We are currently using FEM for deformable simulation. A few caveats we've encountered: 

* FEM is still a bit unstable, occasionally "exploding" on complex objects (3000+ vertices) that have a lot of self-interactions.
* FEM was added sometime last year, and it seems that the material property parameters (e.g. stiffness, damping, etc.) were redirected to FEM backend and thus one cannot change material properties on the older PBD backend without modifying the C++ code. 
* Scale matters. We were struggling to get FEM to work when our objects were in the 0.1~1.0m scale range, when we enlarged everything by 10x, the collision detection and stability improved significantly.
